app/src/androidTest/java/com/duckduckgo/app/browser/downloader/DownloaderUtilTest.kt

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.downloader
+
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class DownloaderUtilTest {
```

Comment:
This contains all issues we discovered so far with various combinations of URL / Content-Disposition / Content-type.